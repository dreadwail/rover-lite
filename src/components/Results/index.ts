export * from './Results';
export { default } from './ResultsContainer';
