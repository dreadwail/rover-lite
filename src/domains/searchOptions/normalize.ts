import {
  BasicOption,
  BooleanOption,
  ChoiceOption,
  DateOption,
  IntegerOption,
  MultipleChoiceOption,
  Option,
  Options,
} from '../../state/searchOptions/types';

interface RawBasicOption {
  readonly type: string;
  readonly required: boolean;
  readonly label: string;
  readonly help_text?: string;
  readonly category?: string;
  readonly category_rank?: number;
  readonly category_display_name?: string;
  readonly rank?: number;
}

interface RawDateOption extends RawBasicOption {
  readonly type: 'date';
}

interface RawChoice {
  readonly value: any;
  readonly display_name: string;
}

interface RawChoiceOption extends RawBasicOption {
  readonly type: 'choice';
  readonly choices: RawChoice[];
}

interface RawMultipleChoiceOption extends RawBasicOption {
  readonly type: 'multiple choice';
  readonly choices: RawChoice[];
}

interface RawIntegerOption extends RawBasicOption {
  readonly type: 'integer';
  readonly min_value: number;
  readonly max_value: number;
}

interface RawBooleanOption extends RawBasicOption {
  readonly type: 'boolean';
}

type RawOption =
  | RawBasicOption
  | RawDateOption
  | RawChoiceOption
  | RawMultipleChoiceOption
  | RawIntegerOption
  | RawBooleanOption;

const isRawBooleanOption = (rawOption: RawOption): rawOption is RawBooleanOption =>
  rawOption.type === 'boolean';
const isRawIntegerOption = (rawOption: RawOption): rawOption is RawIntegerOption =>
  rawOption.type === 'integer';
const isRawMultipleChoiceOption = (rawOption: RawOption): rawOption is RawMultipleChoiceOption =>
  rawOption.type === 'multiple choice';
const isRawChoiceOption = (rawOption: RawOption): rawOption is RawChoiceOption =>
  rawOption.type === 'choice';
const isRawDateOption = (rawOption: RawOption): rawOption is RawDateOption =>
  rawOption.type === 'date';

type OptionsResponseActionType = 'GET';

type RawOptions = Record<string, RawOption>;

export interface OptionsResponse {
  readonly name: string;
  readonly description: string;
  readonly renders: string[];
  readonly parses: string[];
  readonly actions: Record<OptionsResponseActionType, RawOptions>;
}

const extractRawCategory = (rawBasicOption: RawBasicOption): Option['category'] => {
  const defaultCategory: Option['category'] = {
    categoryId: 'detail',
    rank: 0,
    displayName: 'Detail',
  };
  if (!rawBasicOption.category) {
    return defaultCategory;
  }
  if (!rawBasicOption.category_display_name) {
    return defaultCategory;
  }
  if (!rawBasicOption.category_rank) {
    return defaultCategory;
  }
  return {
    categoryId: rawBasicOption.category,
    rank: rawBasicOption.category_rank,
    displayName: rawBasicOption.category_display_name,
  };
};

const labelOverrides: Record<string, string> = {
  Minprice: 'Minimum Price',
  Maxprice: 'Maximum Price',
};

const convertOverridableLabel = (rawLabel: string): string => {
  const override = labelOverrides[rawLabel];
  if (override) {
    return override;
  }
  return rawLabel;
};

const rawBasicOptionToOption = (rawBasicOption: RawBasicOption): BasicOption => ({
  type: rawBasicOption.type,
  required: !!rawBasicOption.required,
  label: convertOverridableLabel(rawBasicOption.label),
  helpText: rawBasicOption.help_text,
  rank: rawBasicOption.rank || 0,
  category: extractRawCategory(rawBasicOption),
});

const rawBooleanOptionToOption = (rawBooleanOption: RawBooleanOption): BooleanOption => {
  const option = rawBasicOptionToOption(rawBooleanOption);
  return { ...option, type: 'boolean' };
};

const rawIntegerOptionToOption = (rawIntegerOption: RawIntegerOption): IntegerOption => {
  const option = rawBasicOptionToOption(rawIntegerOption);
  return {
    ...option,
    type: 'integer',
    minValue: rawIntegerOption.min_value,
    maxValue: rawIntegerOption.max_value,
  };
};

const choiceDisplayNameOverrides: Record<string, string> = {
  no_pref: 'No preference',
  no_other_dogs: 'No other dogs',
  other_dogs: 'Other dogs',
  houses: 'Houses',
  apartments: 'Apartments',
};

const convertOverridableChoiceDisplayName = (rawDisplayName: string): string => {
  const override = choiceDisplayNameOverrides[rawDisplayName];
  if (override) {
    return override;
  }
  return rawDisplayName;
};

const rawChoiceOptionToOption = (rawChoiceOption: RawChoiceOption): ChoiceOption => {
  const option = rawBasicOptionToOption(rawChoiceOption);
  return {
    ...option,
    type: 'choice',
    choices: rawChoiceOption.choices.map(choice => ({
      value: choice.value,
      displayName: convertOverridableChoiceDisplayName(choice.display_name),
    })),
  };
};

const rawMultipleChoiceOptionToOption = (
  rawMultipleChoiceOption: RawMultipleChoiceOption
): MultipleChoiceOption => {
  const option = rawBasicOptionToOption(rawMultipleChoiceOption);
  return {
    ...option,
    type: 'multiple choice',
    choices: rawMultipleChoiceOption.choices.map(choice => ({
      value: choice.value,
      displayName: convertOverridableChoiceDisplayName(choice.display_name),
    })),
  };
};

const rawDateOptionToOption = (rawDateOption: RawDateOption): DateOption => {
  const option = rawBasicOptionToOption(rawDateOption);
  return { ...option, type: 'date' };
};

const rawOptionToOption = (rawOption: RawOption): Option => {
  if (isRawBooleanOption(rawOption)) {
    return rawBooleanOptionToOption(rawOption);
  }
  if (isRawIntegerOption(rawOption)) {
    return rawIntegerOptionToOption(rawOption);
  }
  if (isRawMultipleChoiceOption(rawOption)) {
    return rawMultipleChoiceOptionToOption(rawOption);
  }
  if (isRawChoiceOption(rawOption)) {
    return rawChoiceOptionToOption(rawOption);
  }
  if (isRawDateOption(rawOption)) {
    return rawDateOptionToOption(rawOption);
  }
  return rawBasicOptionToOption(rawOption);
};

const keysToSkip: string[] = ['service_type', 'page'];

export const normalize = (response: OptionsResponse): Options => {
  const rawOptions = response.actions.GET;
  return Object.keys(rawOptions).reduce((options, key) => {
    if (keysToSkip.includes(key)) {
      return options;
    }
    const rawOption = rawOptions[key];
    const option = rawOptionToOption(rawOption);
    return { ...options, [key]: option };
  }, {});
};
