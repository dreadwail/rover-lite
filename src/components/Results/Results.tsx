import React, { Component } from 'react';

import { Listing } from '../../state/results/types';
import ReloadableFailure from '../ReloadableFailure';
import Result from '../Result';
import Spinner from '../Spinner';

export interface ResultsProps {
  readonly isFetching: boolean;
  readonly hasFailed: boolean;
  readonly isStale: boolean;
  readonly listings: Listing[];
  fetchResults(): void;
}

export default class Results extends Component<ResultsProps> {
  public componentDidMount() {
    this.fetchResultsIfNecessary();
  }

  public componentDidUpdate() {
    this.fetchResultsIfNecessary();
  }

  public render() {
    const { isFetching, hasFailed, listings } = this.props;

    if (isFetching) {
      return <Spinner isLargeSize={true} />;
    }

    if (hasFailed) {
      return <ReloadableFailure />;
    }

    return (
      <div className="row">
        <div className="col-sm">
          <ul className="list-group list-group-flush">
            {listings.map(listing => (
              <li key={listing.webUrl} className="list-group-item">
                <Result listing={listing} />
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }

  private readonly fetchResultsIfNecessary = () => {
    const { fetchResults, isStale, isFetching } = this.props;
    const needsFetch = isStale && !isFetching;
    if (needsFetch) {
      fetchResults();
    }
  };
}
