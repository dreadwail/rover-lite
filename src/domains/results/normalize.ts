import { Listing } from '../../state/results/types';

export interface RawDerivedData {
  readonly zip_code: string;
}

export interface RawResponsiveness {
  readonly photo_rate?: number;
  readonly photo_rate_desc: string;
  readonly percent?: number;
  readonly percent_desc: string;
  readonly time_short_desc: string;
}

export interface RawImage {
  readonly small: string;
}

export interface RawAttribute {
  readonly text: string;
  readonly key: string;
  readonly icon: string;
}

export interface RawPreferences {
  readonly dogs_preferences_attributes: RawAttribute[];
  readonly accepted_dogs_attributes: RawAttribute[];
  readonly dogs_experience_attributes: RawAttribute[];
  readonly attributes: RawAttribute[];
}

export interface RawProviderProfile {
  readonly hosting_service_preferences: RawPreferences;
  readonly traveling_service_preferences: RawPreferences;
  readonly attributes: RawAttribute[];
  readonly years_of_experience: number;
}

export interface RawBadge {
  readonly icon: string;
  readonly description: string;
  readonly title: string;
}

export interface RawPersonSummary {
  readonly responsiveness: RawResponsiveness;
  readonly first_name: string;
  readonly reviews_count: number;
  readonly profile_photo: RawImage;
  readonly ratings_average: string;
  readonly web_url: string;
  readonly images: RawImage[];
  readonly provider_profile: RawProviderProfile;
  readonly attributes: Record<string, RawAttribute>;
  readonly badges: RawBadge[];
}

export interface RawBadgeData {
  readonly content: string;
  readonly image_url: string;
  readonly show_on_search: boolean;
  readonly title: string;
}

export interface RawReview {
  readonly content: string;
  readonly dog_photo_url: string;
}

export interface RawResult {
  readonly rank: number;
  readonly person_summary: RawPersonSummary;
  readonly reviews_count: number;
  readonly neighborhood: string;
  readonly price_unit: string;
  readonly city: string;
  readonly zip: string;
  readonly title: string;
  readonly state: string;
  readonly contact_url: string;
  readonly person_name: string;
  readonly price: string;
  readonly review_text: string;
  readonly review_dog_photo_url: string;
  readonly badge_data: RawBadgeData[];
  readonly repeat_client_count: number;
  readonly reviews: RawReview[];
  readonly ratings_average: string;
  readonly currency_symbol: string;
}

export interface RawSearchResponse {
  readonly results: RawResult[];
}

const rawResultToListing = (rawResult: RawResult): Listing => ({
  thumbnailUrl: rawResult.person_summary.profile_photo.small,
  rank: rawResult.rank,
  name: rawResult.person_summary.first_name,
  badges: rawResult.badge_data.map(rawBadge => ({
    content: rawBadge.content,
    imageUrl: rawBadge.image_url,
    title: rawBadge.title,
  })),
  tagline: rawResult.title,
  neighborhood: rawResult.neighborhood,
  city: rawResult.city,
  state: rawResult.state,
  zip: rawResult.zip,
  repeatClientCount: rawResult.repeat_client_count,
  reviewCount: rawResult.reviews_count,
  ratingAverage: parseFloat(rawResult.ratings_average),
  featuredReview: rawResult.review_text,
  currencySymbol: rawResult.currency_symbol,
  price: parseFloat(rawResult.price),
  priceUnit: rawResult.price_unit,
  webUrl: rawResult.person_summary.web_url,
});

export const normalize = (rawResponse: RawSearchResponse): Listing[] =>
  rawResponse.results.map(rawResultToListing);
