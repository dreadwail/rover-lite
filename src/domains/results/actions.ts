import { Action } from 'redux';

import { request } from '../../api/http';
import { urlEncodeQuery } from '../../api/query';
import { Listing } from '../../state/results/types';
import { SearchInputs } from '../../state/search/types';
import debouncedAction from '../debouncedThunk';
import { getInputs, getLatLong, getSearchType } from '../search/selectors';
import { PromiseThunkAction } from '../types';

import { normalize, RawSearchResponse } from './normalize';

export interface FetchResultsPendingAction extends Action {
  readonly type: 'FETCH_RESULTS.PENDING';
}

export interface FetchResultsFulfilledAction extends Action {
  readonly type: 'FETCH_RESULTS.FULFILLED';
  readonly payload: Listing[];
}

export interface FetchResultsRejectedAction extends Action {
  readonly type: 'FETCH_RESULTS.REJECTED';
  readonly payload: {
    readonly error: any;
  };
  readonly error: true;
}

const fetchResultsPending = (): FetchResultsPendingAction => ({
  type: 'FETCH_RESULTS.PENDING',
});

const fetchResultsFulfilled = (listings: Listing[]): FetchResultsFulfilledAction => ({
  type: 'FETCH_RESULTS.FULFILLED',
  payload: listings,
});

const fetchResultsRejected = (error: any): FetchResultsRejectedAction => ({
  type: 'FETCH_RESULTS.REJECTED',
  payload: {
    error,
  },
  error: true,
});

export type FetchResults = () => PromiseThunkAction<void>;
export const fetchResults: FetchResults = () => async (dispatch, getState) => {
  dispatch(fetchResultsPending());
  try {
    const state = getState();

    const searchType = getSearchType(state);
    const [latitude, longitude] = getLatLong(state);

    const defaultInputs: SearchInputs = {
      centerlat: latitude,
      centerlng: longitude,
      service_type: searchType,
    };
    const userInputs = getInputs(state);
    const inputs = { ...defaultInputs, ...userInputs };
    const encodedQuery = urlEncodeQuery(inputs);

    const urlBase = 'https://www.rover.com/api/v3/search';
    const url = `${urlBase}/${searchType}/?${encodedQuery}`;

    const response = await request<RawSearchResponse>('GET', url);

    const normalizedListings = normalize(response);
    dispatch(fetchResultsFulfilled(normalizedListings));
  } catch (err) {
    dispatch(fetchResultsRejected(err));
  }
};

export const debouncedFetchResults = debouncedAction(fetchResults, 250);
