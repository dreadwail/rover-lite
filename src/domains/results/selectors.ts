import { Listing } from '../../state/results/types';
import { RoverSearchState } from '../../state/types';

export const isFetching = (state: RoverSearchState): boolean => state.results.isFetching;
export const hasFailed = (state: RoverSearchState): boolean => state.results.hasFailed;
export const isStale = (state: RoverSearchState): boolean => state.results.isStale;
export const getListings = (state: RoverSearchState): Listing[] => state.results.listings;
