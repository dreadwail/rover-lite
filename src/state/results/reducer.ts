import { Reducer } from 'redux';

import {
  FetchResultsFulfilledAction,
  FetchResultsPendingAction,
  FetchResultsRejectedAction,
} from '../../domains/results/actions';
import { SearchFieldChangedAction, SearchTypeChangedAction } from '../../domains/search/actions';

import { Results } from './types';

export type ResultsAction =
  | FetchResultsPendingAction
  | FetchResultsFulfilledAction
  | FetchResultsRejectedAction
  | SearchTypeChangedAction
  | SearchFieldChangedAction;

const initialState: Results = {
  isFetching: false,
  hasFailed: false,
  isStale: true,
  listings: [],
};

const reducer: Reducer<Results, ResultsAction> = (state = initialState, action) => {
  switch (action.type) {
    case 'FETCH_RESULTS.PENDING':
      return { ...state, isFetching: true, hasFailed: false, isStale: false };
    case 'FETCH_RESULTS.FULFILLED':
      return {
        ...state,
        isFetching: false,
        hasFailed: false,
        isStale: false,
        listings: action.payload,
      };
    case 'FETCH_RESULTS.REJECTED':
      return { ...state, isFetching: false, hasFailed: true };
    case 'SEARCH.TYPE_CHANGED':
      return { ...state, isStale: true };
    case 'SEARCH.FIELD_CHANGED':
      return { ...state, isStale: true };
    default: {
      return state;
    }
  }
};

export default reducer;
