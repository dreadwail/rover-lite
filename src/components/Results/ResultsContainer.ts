import { connect, MapStateToProps } from 'react-redux';

import { debouncedFetchResults } from '../../domains/results/actions';
import { getListings, hasFailed, isFetching, isStale } from '../../domains/results/selectors';
import { MapDispatchToProps } from '../../domains/types';
import { RoverSearchState } from '../../state/types';

import Results, { ResultsProps } from './Results';

export type OwnProps = {};
export type MappedStateProps = Pick<
  ResultsProps,
  'isFetching' | 'hasFailed' | 'isStale' | 'listings'
>;
export type MappedDispatchProps = Pick<ResultsProps, 'fetchResults'>;

export const mapStateToProps: MapStateToProps<
  MappedStateProps,
  OwnProps,
  RoverSearchState
> = state => ({
  isFetching: isFetching(state),
  hasFailed: hasFailed(state),
  isStale: isStale(state),
  listings: getListings(state),
});

export const mapDispatchToProps: MapDispatchToProps<MappedDispatchProps, OwnProps> = dispatch => ({
  fetchResults: async () => {
    await dispatch(debouncedFetchResults());
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Results);
