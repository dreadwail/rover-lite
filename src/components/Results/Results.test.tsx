import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { normalize } from '../../domains/results/normalize';
import overnightBoarding from '../../domains/results/__fixtures__/overnight-boarding.json';

import Results, { ResultsProps } from './Results';

describe('Results', () => {
  let props: ResultsProps;
  let rendered: ShallowWrapper<ResultsProps>;

  beforeEach(() => {
    const [first, second] = normalize(overnightBoarding);
    props = {
      isFetching: false,
      hasFailed: false,
      isStale: false,
      listings: [first, second],
      fetchResults: jest.fn().mockName('fetchResults'),
    };
  });

  describe('when fetching', () => {
    beforeEach(() => {
      props = { ...props, isFetching: true };
      rendered = shallow(<Results {...props} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when failed', () => {
    beforeEach(() => {
      props = { ...props, hasFailed: true };
      rendered = shallow(<Results {...props} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('when the results are not stale and not fetching', () => {
    beforeEach(() => {
      props = { ...props, isStale: false, isFetching: false };
      (props.fetchResults as jest.Mock).mockClear();
      rendered = shallow(<Results {...props} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });

    it('does not call fetchResults on mount', () => {
      expect(props.fetchResults).not.toHaveBeenCalled();
    });

    describe('when the results become stale', () => {
      beforeEach(() => {
        (props.fetchResults as jest.Mock).mockClear();
        rendered.setProps({ isStale: true });
      });

      it('calls fetchResults', () => {
        expect(props.fetchResults).toHaveBeenCalled();
      });
    });
  });

  describe('when the results are stale and not fetching', () => {
    beforeEach(() => {
      props = { ...props, isStale: true, isFetching: false };
      (props.fetchResults as jest.Mock).mockClear();
      rendered = shallow(<Results {...props} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });

    it('calls fetchResults on mount', () => {
      expect(props.fetchResults).toHaveBeenCalled();
    });
  });

  describe('when the results are not stale but fetching', () => {
    beforeEach(() => {
      props = { ...props, isStale: false, isFetching: true };
      (props.fetchResults as jest.Mock).mockClear();
      rendered = shallow(<Results {...props} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });

    it('does not call fetchResults on mount', () => {
      expect(props.fetchResults).not.toHaveBeenCalled();
    });
  });
});
