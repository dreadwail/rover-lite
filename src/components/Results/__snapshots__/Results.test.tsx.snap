// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results when failed renders correctly 1`] = `<ReloadableFailure />`;

exports[`Results when fetching renders correctly 1`] = `
<Spinner
  isLargeSize={true}
/>
`;

exports[`Results when the results are not stale and not fetching renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
        key="https://www.rover.com/members/madison-s-2-pet-loversfenced-yard-in-ballard/"
      >
        <Result
          listing={
            Object {
              "badges": Array [
                Object {
                  "content": "This sitter has successfully passed an enhanced <a href=\\"/background-checks/\\">background check</a> by a third party provider.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/verified-enhanced-background-check/icons/icon_21e9a29c.png",
                  "title": "Verified Enhanced Background Check",
                },
                Object {
                  "content": "This sitter has completed <a href=\\"https://www.rover.com/sitter-training\\">Rover 101</a> for pet care best practices.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/rover-training-graduate/icons/icon_1357faa9.png",
                  "title": "Rover 101",
                },
              ],
              "city": "Seattle",
              "currencySymbol": "$",
              "featuredReview": "Best dog sitter I've ever used. Easy-going and accommodating, was eager to figure out schedule and communication. My dog was so happy with them, she forgot I was gone. I'll be using them every time so long as they're available!",
              "name": "Madison",
              "neighborhood": "Loyal Heights",
              "price": 45,
              "priceUnit": "night",
              "rank": 1,
              "ratingAverage": 5,
              "repeatClientCount": 6,
              "reviewCount": 11,
              "state": "WA",
              "tagline": "2 pet lovers+fenced yard in Ballard",
              "thumbnailUrl": "https://rover-com.imgix.net/images/people/3gmEo9Rg/blrepvpeti/original.png?crop=facecenter&fit=crop&h=200&ixlib=python-1.2.0&w=200",
              "webUrl": "https://www.rover.com/members/madison-s-2-pet-loversfenced-yard-in-ballard/",
              "zip": "98117",
            }
          }
        />
      </li>
      <li
        className="list-group-item"
        key="https://www.rover.com/members/angie-b-i-absolutely-love-dogs/"
      >
        <Result
          listing={
            Object {
              "badges": Array [
                Object {
                  "content": "This sitter has successfully passed an enhanced <a href=\\"/background-checks/\\">background check</a> by a third party provider.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/verified-enhanced-background-check/icons/icon_21e9a29c.png",
                  "title": "Verified Enhanced Background Check",
                },
              ],
              "city": "Seattle",
              "currencySymbol": "$",
              "featuredReview": "We boarded our Havanese, Ryley, with Angie, while we were away for a long weekend.  We could not have been more pleased.  Angie is caring and considerate, responsible and prompt with all communications.  And all indications are that Ryley had a wonder time with Angie, and visa-versa, which is exactly what we had hoped!",
              "name": "Angie",
              "neighborhood": "Olympic Manor",
              "price": 35,
              "priceUnit": "night",
              "rank": 2,
              "ratingAverage": 5,
              "repeatClientCount": 2,
              "reviewCount": 9,
              "state": "WA",
              "tagline": "I absolutely love dogs!!!",
              "thumbnailUrl": "https://rover-com.imgix.net/images/people/g45DYroQ/xuouqxoook/original.jpg?crop=facecenter&fit=crop&h=200&ixlib=python-1.2.0&w=200",
              "webUrl": "https://www.rover.com/members/angie-b-i-absolutely-love-dogs/",
              "zip": "98117",
            }
          }
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Results when the results are not stale but fetching renders correctly 1`] = `
<Spinner
  isLargeSize={true}
/>
`;

exports[`Results when the results are stale and not fetching renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
        key="https://www.rover.com/members/madison-s-2-pet-loversfenced-yard-in-ballard/"
      >
        <Result
          listing={
            Object {
              "badges": Array [
                Object {
                  "content": "This sitter has successfully passed an enhanced <a href=\\"/background-checks/\\">background check</a> by a third party provider.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/verified-enhanced-background-check/icons/icon_21e9a29c.png",
                  "title": "Verified Enhanced Background Check",
                },
                Object {
                  "content": "This sitter has completed <a href=\\"https://www.rover.com/sitter-training\\">Rover 101</a> for pet care best practices.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/rover-training-graduate/icons/icon_1357faa9.png",
                  "title": "Rover 101",
                },
              ],
              "city": "Seattle",
              "currencySymbol": "$",
              "featuredReview": "Best dog sitter I've ever used. Easy-going and accommodating, was eager to figure out schedule and communication. My dog was so happy with them, she forgot I was gone. I'll be using them every time so long as they're available!",
              "name": "Madison",
              "neighborhood": "Loyal Heights",
              "price": 45,
              "priceUnit": "night",
              "rank": 1,
              "ratingAverage": 5,
              "repeatClientCount": 6,
              "reviewCount": 11,
              "state": "WA",
              "tagline": "2 pet lovers+fenced yard in Ballard",
              "thumbnailUrl": "https://rover-com.imgix.net/images/people/3gmEo9Rg/blrepvpeti/original.png?crop=facecenter&fit=crop&h=200&ixlib=python-1.2.0&w=200",
              "webUrl": "https://www.rover.com/members/madison-s-2-pet-loversfenced-yard-in-ballard/",
              "zip": "98117",
            }
          }
        />
      </li>
      <li
        className="list-group-item"
        key="https://www.rover.com/members/angie-b-i-absolutely-love-dogs/"
      >
        <Result
          listing={
            Object {
              "badges": Array [
                Object {
                  "content": "This sitter has successfully passed an enhanced <a href=\\"/background-checks/\\">background check</a> by a third party provider.",
                  "imageUrl": "//dt0499hd9onpe.cloudfront.net/badges/verified-enhanced-background-check/icons/icon_21e9a29c.png",
                  "title": "Verified Enhanced Background Check",
                },
              ],
              "city": "Seattle",
              "currencySymbol": "$",
              "featuredReview": "We boarded our Havanese, Ryley, with Angie, while we were away for a long weekend.  We could not have been more pleased.  Angie is caring and considerate, responsible and prompt with all communications.  And all indications are that Ryley had a wonder time with Angie, and visa-versa, which is exactly what we had hoped!",
              "name": "Angie",
              "neighborhood": "Olympic Manor",
              "price": 35,
              "priceUnit": "night",
              "rank": 2,
              "ratingAverage": 5,
              "repeatClientCount": 2,
              "reviewCount": 9,
              "state": "WA",
              "tagline": "I absolutely love dogs!!!",
              "thumbnailUrl": "https://rover-com.imgix.net/images/people/g45DYroQ/xuouqxoook/original.jpg?crop=facecenter&fit=crop&h=200&ixlib=python-1.2.0&w=200",
              "webUrl": "https://www.rover.com/members/angie-b-i-absolutely-love-dogs/",
              "zip": "98117",
            }
          }
        />
      </li>
    </ul>
  </div>
</div>
`;
